examples/ota-requestor-app/ota-requestor-common/ExampleOTARequestor.cpp

```diff
@@ -91,8 +91,16 @@ EmberAfStatus ExampleOTARequestor::HandleAnnounceOTAProvider(
     mProviderNodeId      = providerLocation;
     mProviderFabricIndex = commandObj->GetExchangeContext()->GetSessionHandle().GetFabricIndex();
 
-    ChipLogProgress(SoftwareUpdate, "Notified of Provider at NodeID: 0x" ChipLogFormatX64 " on FabricIndex 0x%" PRIu8,
-                    ChipLogValueX64(mProviderNodeId), mProviderFabricIndex);
+    ChipLogProgress(SoftwareUpdate, "OTA Requestor received AnnounceOTAProvider");
+    ChipLogDetail(SoftwareUpdate, "  FabricIndex: %" PRIu8, mProviderFabricIndex);
+    ChipLogDetail(SoftwareUpdate, "  ProviderNodeID: 0x" ChipLogFormatX64, ChipLogValueX64(mProviderNodeId));
+    ChipLogDetail(SoftwareUpdate, "  VendorID: 0x%" PRIx16, commandData.vendorId);
+    ChipLogDetail(SoftwareUpdate, "  AnnouncementReason: %" PRIu8, announcementReason);
+    if (commandData.metadataForNode.HasValue())
+    {
+        ChipLogDetail(SoftwareUpdate, "  MetadataForNode: %.*s", static_cast<int>(commandData.metadataForNode.Value().size()),
+                      commandData.metadataForNode.Value().data());
```

Comment:
MetadataForNode is an octet string, so may contain non-printable characters.  This should be dumped out as hex.

Confusingly, the contents of the field is a List:

> This optional field, if present, SHALL consist of a top-level anonymous list; each list element SHALL have a profile-specific tag encoded in fully-qualified form.

So apparently, it's not opaque data, but a list of opaque elements.  Either way, it can't be printed like this.

Reply:
Thanks for the catch. Just verified with the tester that these fields aren't checked as they are vendor specific so modifying to match `metadataForRequestor` for the auto generated print out for `QueryImageResponse` to indicate size only.

Reply:
Sure, that works.